app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt

```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
 
     @Provides
     @Singleton
-    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage = AndroidKeyStorage(context)
+    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage =
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
```

Comment:
Do we need to do the API level check? I suppose we could just always use `Locale.ENGLISH` specifically for `KeyStorage` as well, then there would be fewer surprises.

Reply:
Yes. We could. My idea was to remove this check and the branch, when have minSdkVersion > M.

Reply:
Would it be better to always use `Locale.English` independently of the SDK version for `bouncy_castle`? Having this small fork in logic here could be problematic as we would have small changes in behaviour depending on the device.

Reply:
My risk assessment is like this: 
- If the locale switching turns out to be a problem, then it is only a problem on a minority of devices )if we have this branch logic)
- Google considered this a bug and fixed it, so it should not occur in devices after M

Reply:
That makes sense. I understand now how we are better covered with this approach. Thanks!